### DISCUSSION TITLE
Support the Navigation API

### DISCUSSION THREAD

**Discussion by loganvolkers** [TanStack/router#821] [General]
### What problem does this feature solve?
Chromium 102 has been released stable, which means that Chrome, Edge, Opera, Brave, Vivaldi, etc and Electron supports the Navigation API https://chromestatus.com/feature/6232287446302720 The [Navigation API](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API) is the successor of the History API and is better suited for SPAs. Therefore, routers should allow to target and leverage it.

 ### Why should this feature be included?
 > The existing window.history API is hard to deal with in practice, especially for single-page applications. In the best case, developers can work around this with various hacks. In the worst case, it causes user-facing pain in the form of lost state and broken back buttons, or the inability to achieve the desired navigation flow for a web app.

### What does the proposed API look like?
see the spec https://chromestatus.com/feature/6232287446302720

### Where can I learn more?
Being discussed in the interop project: https://github.com/web-platform-tests/interop/issues/435

**Comment by LubosAngus** [comment]
It's available in all major browsers since January 2026. 

Any updates on the plans for implementing this into TanStack router?

https://developer.mozilla.org/en-US/docs/Web/API/Navigation
